### PR TITLE
(BKR-439) Add DNF for fedora-22

### PR DESCRIPTION
**Pull request by johnduarte**
This commit adds a case condition for fedora-22 to use the 'dnf'
command where the 'yum' command would previously have been executed.

### CODE REVIEW

File: lib/beaker/dsl/install_utils/foss_utils.rb [hunk 1]
```diff
@@ -1231,6 +1231,9 @@ def install_puppet_agent_pe_promoted_repo_on( hosts, opts )
             scp_to host, File.join(copy_dir_local, download_file), onhost_copy_base
 
             case variant
+            when /^(fedora-22)$/
+              on host, "tar -zxvf #{onhost_copied_download} -C #{onhost_copy_base}"
+              on host, "dhf --nogpgcheck localinstall -y #{onhost_copied_file}"
```
Comment (er0ck) [inline lib/beaker/dsl/install_utils/foss_utils.rb]: typo in 'dnf' here?

**Comment by er0ck** [comment]
no tests?

**Comment by johnduarte** [comment]
The underlying `rpm` call should be the same and is covered by

https://github.com/puppetlabs/beaker/blob/master/spec/beaker/host/unix/pkg_spec.rb#L41

**Comment by electrical** [comment]
A test should be added so we can confirm the right command us used for fedora 22.

**Comment by kevpl** [comment]
last comment (from electrical) still current

File: lib/beaker/dsl/install_utils/foss_utils.rb [hunk 2]
```diff
@@ -1231,6 +1231,9 @@ def install_puppet_agent_pe_promoted_repo_on( hosts, opts )
             scp_to host, File.join(copy_dir_local, download_file), onhost_copy_base
 
             case variant
+            when /^(fedora-22)$/
+              on host, "tar -zxvf #{onhost_copied_download} -C #{onhost_copy_base}"
```
Comment (er0ck) [inline lib/beaker/dsl/install_utils/foss_utils.rb]: all these tars for non-windows should/could probably be pulled out of the case statement

**Comment by johnduarte** [comment]
Spec test added to confirm use of `dnf` on `fedora-22` and `yum` on `fedora-20`

**Comment by kevpl** [comment]
:100:

**Pull request merged**